### ISSUE TITLE
Source mapping failed: A source map could not be applied

### ISSUE THREAD

**Issue by guilhermedeandrade** [bugsnag/bugsnag-react-native#474]
### Description

The stack trace seems fine, but I got this warning: 

`Source mapping failed: A source map could not be applied`

Is something really not working as expected?

![image](https://user-images.githubusercontent.com/8797405/89309821-d2850880-d66b-11ea-83d9-f9272ec1b03d.png)

### Issue

Source mapping failed: A source map could not be applied

### Environment

Library versions:

```shell
react-native-with-moxy@0.0.0 /Users/guilherme/workspace/moxy/react-native-with-moxy
├── bugsnag-react-native@2.23.10 
└── react-native@0.61.5 
```

- cocoapods version (if any) (`pod --version`):
- iOS/Android version(s): Android 10
- simulator/emulator or physical device?: simulator
- debug mode or production?: production

- [ ] (iOS only) `[BugsnagReactNative start]` is present in the
  `application:didFinishLaunchingWithOptions:` method in your `AppDelegate`
  class?
- [ ] (Android only) `BugsnagReactNative.start(this)` is present in the
  `onCreate` method of your `MainApplication` class? No

**Comment by mattdyoung** [comment]
@guilhermedeandrade 
The stackframe I can see in your screenshot looks like it has successfully had the source map applied.

Are there other frames in the stacktrace which have failed to map? Stackframes we determine are not 'in-project' are hidden by are shown when you select 'Full trace'.

**Comment by guilhermedeandrade** [comment]
> @guilhermedeandrade
> The stackframe I can see in your screenshot looks like it has successfully had the source map applied.
> 
> Are there other frames in the stacktrace which have failed to map? Stackframes we determine are not 'in-project' are hidden by default but are shown when you select 'Full trace'.

Exactly, the stackframe shown in the screenshot really looks like it has successfully had the source map applied. Here is the screenshot when I select "Full trace":

![image](https://user-images.githubusercontent.com/8797405/89374648-ebc99b80-d6e3-11ea-9804-0f3e124dd1c7.png)

**Comment by xander-jones** [comment]
Hey @guilhermedeandrade, in this event there are 4 frames in the stacktrace. The first one is the only 'in-project' frame, which is the frame most of interest here, and that's mapped successfully. The following 2 frames have also been mapped correctly. If you expand the final frame (which reads `global code`), you'll notice that this line points to `index.android.bundle:836:4` which is :
```
//# sourceMappingURL=android-release.bundle.map
```
The failure details are: Source map did not contain mapping for frame `index.android.bundle:836:4`, which is unsurprising as this is not minified code. 

It looks like a lot of your events have this final frame being sent through. Are you modifying the stacktrace?

**Comment by guilhermedeandrade** [comment]
> Hey @guilhermedeandrade, in this event there are 4 frames in the stacktrace. The first one is the only 'in-project' frame, which is the frame most of interest here, and that's mapped successfully. The following 2 frames have also been mapped correctly. If you expand the final frame (which reads `global code`), you'll notice that this line points to `index.android.bundle:836:4` which is :
> 
> ```
> //# sourceMappingURL=android-release.bundle.map
> ```
> 
> The failure details are: Source map did not contain mapping for frame `index.android.bundle:836:4`, which is unsurprising as this is not minified code.
> 
> It looks like a lot of your events have this final frame being sent through. Are you modifying the stacktrace?

Thanks for your explanation @xander-jones, it helped me to understand the situation! I'm not modifying the stacktrace though.

**Comment by xander-jones** [comment]
Hey @guilhermedeandrade, no problem! On further inspection, it looks like this `global code` reference is something that we've seen before (albeit rare). It's something that we'll be taking a look into. In the meantime, you're not doing anything incorrectly, and the frames of interest that contain actual code are being unminified correctly. We'll keep you updated with any changes!
